app/routes.py

```diff
@@ -1 +1,235 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, request, abort, make_response
+from app import db, Key
+from app.models.task import Task
+from app.models.goal import Goal
+import datetime
+import requests
+
+
+task_bp = Blueprint("task_bp", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+#helper function to validate that object ids are valid
+def validate_id(cls, obj_id):
+    try:
+        obj_id = int(obj_id)
+    except ValueError:
+        response_str = f"Invalid {cls.__name__}_id: {obj_id} must be an integer"
+        abort(make_response(jsonify({"message":response_str}), 400))
+
+    matching_obj = cls.query.get(obj_id)
+
+    if not matching_obj:
+        response_str = f"Could not find {cls.__name__} with id {obj_id}"
+        abort(make_response(jsonify({"message":response_str}), 404))
+    
+    return matching_obj
+
+
+@task_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body:
+        abort(make_response(jsonify({"details":"Invalid data"}), 400))
+    
+    else:
+        new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"])
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        new_task_dict = {"task": new_task.make_task_dict()}
+        return jsonify(new_task_dict), 201
+
+@task_bp.route("", methods=["GET"])
+def get_all_tasks():
+    tasks = Task.query.all()
+    sort_param = request.args.get("sort")
+
+
+    response = []
+    for task in tasks:
+        task_dict = task.make_task_dict()
+        response.append(task_dict)
+    
+    if sort_param == "asc":
+        response = sorted(response, key=lambda i:i['title'])
+    if sort_param == "desc":
+        response = sorted(response, key=lambda i:i['title'], reverse=True)
```

Comment:
This works, but it's more efficient if you let postgres do the sorting.  SQL Alchemy has some methods to let Postgres do the sorting.

https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.order_by